### DISCUSSION TITLE
Changing internals underbars

### DISCUSSION THREAD

**Discussion by rumbledethumps** [cc65/cc65#2261] [General]
I maintain the SDK for my project external to CC65. It's been working fine except for one problem. I'm getting support requests because critical symbols are changing. I can't find any obvious mechanism for backwards compatibility.

I worked around osmaperrno changing by exporting both symbols.
```
__osmaperrno
___osmaperrno
```
I need some ideas for the mappederrno change.  Is there something reliable I can switch on for these changes, like a version number?
```
__mappederrno
___mappederrno
```

**Comment by kugelfuhr** [comment]
Yeah, that was a bit unfortunate - especially since only two of the changes were actually necessary. But the developers decided to be on the safe side by changing them all, which is understandable.

I don't know your exact use case, so the following two ideas may help - or not:

You might open an issue and request a symbol in the assembler includes plus a macro in the standard headers that lets you check for the changes names.

You can also create compatibility modules that contain something like:
```
        .import  ___errno
        .export  __errno := ___errno
        .import  ___mappederrno
        .export  __mappederrno := ___mappederrno
```
It allows code written for older versions to link against new versions by translating the identifiers. All that needs to be done is to link in the module. 
A similar module may be used to have the changes the other way round:
```
        .import  __errno
        .export  ___errno :=___errno
        .import  __mappederrno
        .export  ___mappederrno := __mappederrno
```
This module allows code written for a newer compiler to use old libraries.

**Comment by rumbledethumps** [comment]
It might be time to add a new target:
https://github.com/cc65/cc65/pull/2268
Having the SDK part of CC65 should keep this from happening again in the future.
